chia-tools/src/bin/run-spend.rs

```diff
@@ -0,0 +1,285 @@
+use chia::gen::conditions::Condition;
+use chia_protocol::Bytes32;
+use chia_traits::Streamable;
+use clap::Parser;
+use clvm_traits::{FromClvm, ToClvm};
+use clvm_utils::tree_hash;
+use clvm_utils::CurriedProgram;
+use clvmr::{allocator::NodePtr, Allocator};
+use hex_literal::hex;
+use std::io::Cursor;
+
+/// Run a puzzle given a solution and print the resulting conditions
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// Path to CoinSpend (serialized binary file)
+    spend: String,
+}
+
+trait DebugPrint {
+    fn debug_print(&self, a: &Allocator) -> String;
+}
+
+impl DebugPrint for NodePtr {
+    fn debug_print(&self, a: &Allocator) -> String {
+        hex::encode(a.atom(*self))
+    }
+}
+
+impl DebugPrint for Condition {
```

Comment:
Seems like this could be generally useful and you might want to move it to a separate file.

Reply:
yeah, it would be great to be a trait and to have a derive macro. I'm a bit hesitant to expand the scope of this PR though. I think it would make sense to punt on that until the need for a facility like this comes up again